react/javascript/src/components/app/ExecutionSummary.tsx

```diff
@@ -1,86 +1,127 @@
-  return testRunStarted ? testRunStarted.testRunStarted : undefined
-}
-
-function findTestRunFinished(envelopesQuery: EnvelopesQuery): messages.TestRunFinished {
-  const testRunFinished = envelopesQuery.find((envelope) => envelope.testRunFinished !== null)
-  return testRunFinished ? testRunFinished.testRunFinished : undefined
-}
-
-interface IProps {
+export interface IExecutionSummaryProps {
+  scenarioCountByStatus: Map<messages.TestStepResultStatus, number>
+  totalScenarioCount: number
+  testRunStarted: TestRunStarted
+  testRunFinished: TestRunFinished
+  referenceDate?: Date
   meta: messages.Meta
 }
 
-export const ExecutionSummary: React.FunctionComponent<IProps> = ({ meta: meta }) => {
-  const envelopesQuery = React.useContext(EnvelopesQueryContext)
-
-  const testRunStarted = findTestRunStarted(envelopesQuery)
-  const testRunFinished = findTestRunFinished(envelopesQuery)
-  const millisDuration = getDurationsMillis(testRunStarted, testRunFinished)
-
+export const ExecutionSummary: React.FunctionComponent<IExecutionSummaryProps> = ({
+  scenarioCountByStatus,
+  totalScenarioCount,
+  testRunStarted,
+  testRunFinished,
+  referenceDate,
+  meta,
+}) => {
+  const percentagePassed: string =
+    new Intl.NumberFormat(undefined, {
+      style: 'percent',
+    }).format((scenarioCountByStatus.get(TestStepResultStatus.PASSED) ?? 0) / totalScenarioCount) +
+    ' passed'
+  const startDate: Date = new Date(
+    TimeConversion.timestampToMillisecondsSinceEpoch(testRunStarted.timestamp)
+  )
+  const finishDate: Date = new Date(
+    TimeConversion.timestampToMillisecondsSinceEpoch(testRunFinished.timestamp)
+  )
+  const formattedTimestamp: string = formatDistanceStrict(startDate, referenceDate ?? new Date(), {
+    addSuffix: true,
+  })
+  const formattedDuration: string = formatDurationNicely(startDate, finishDate)
   return (
-    <div className="cucumber-execution-data">
-      <table>
-        <tbody>
-          {millisDuration && (
-            <tr>
-              <th>Duration</th>
-              <td>
-                <Duration durationMillis={millisDuration} />
-              </td>
-            </tr>
-          )}
-          {meta.ci && (
-            <tr>
-              <th>Build</th>
-              <td>
-                <a href={meta.ci.url}>{meta.ci.name}</a>
-              </td>
-            </tr>
-          )}
-          {meta.ci && (
-            <tr>
-              <th>Commit</th>
-              <td>
-                <CICommitLink ci={meta.ci} />
-              </td>
-            </tr>
-          )}
-          {meta.implementation && <Product name="Implementation" product={meta.implementation} />}
-          {meta.runtime && <Product name="Runtime" product={meta.runtime} />}
-          {meta.os && <Product name="OS" product={meta.os} />}
-          {meta.cpu && <Product name="CPU" product={meta.cpu} />}
-        </tbody>
-      </table>
+    <div className={styles.backdrop}>
```

Comment:
I really like that all the `classNames` are external to the tsx! I suppose that means consumers that use tailwind for the rest of the app can define those classes like this?

```css
.list {
  @apply ... tailwind classes ...
}
```

Reply:
For the builtin styling, the class name that actually comes out will be a unique generated string - this is a nice feature of CSS modules gets around the global namespace problem of CSS (esp when we're in someone else's app).

For components with custom rendering support, when users provide their own class names they can be whatever they want, including class names that reference a separate stylesheet, ones imported from another CSS module, or even the tailwind utility class names directly.

```tsx
<CucumberReact
        customRendering={{
          Tags: {
            tags: 'custom-list-class',
            tag: 'custom-item-class',
          },
        }}
      >
```

(I haven't actually made `<ExecutionSummary/>` et all customisable as part of this - happy to do another PR if you're going to need that though!)